File: clients/node/sdk/models/searchResult.ts
import {createAccountFromDiscriminatorValue} from './createAccountFromDiscriminatorValue';
import {createStatusFromDiscriminatorValue} from './createStatusFromDiscriminatorValue';
import {createTagFromDiscriminatorValue} from './createTagFromDiscriminatorValue';
import {Account, Status, Tag} from './index';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class SearchResult implements AdditionalDataHolder, Parsable {
    /** The accounts property */
    private _accounts?: Account[] | undefined;
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The hashtags property */
    private _hashtags?: Tag[] | undefined;
    /** The statuses property */
    private _statuses?: Status[] | undefined;
    /**
     * Gets the accounts property value. The accounts property
     * @returns a Account
     */
    public get accounts() {
        return this._accounts;
    };
    /**
     * Sets the accounts property value. The accounts property
     * @param value Value to set for the accounts property.
     */
    public set accounts(value: Account[] | undefined) {
        this._accounts = value;
    };
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Instantiates a new SearchResult and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "accounts": n => { this.accounts = n.getCollectionOfObjectValues<Account>(createAccountFromDiscriminatorValue); },
            "hashtags": n => { this.hashtags = n.getCollectionOfObjectValues<Tag>(createTagFromDiscriminatorValue); },
            "statuses": n => { this.statuses = n.getCollectionOfObjectValues<Status>(createStatusFromDiscriminatorValue); },
        };
    };
    /**
     * Gets the hashtags property value. The hashtags property
     * @returns a Tag
     */
    public get hashtags() {
        return this._hashtags;
    };
    /**
     * Sets the hashtags property value. The hashtags property
     * @param value Value to set for the hashtags property.
     */
    public set hashtags(value: Tag[] | undefined) {
        this._hashtags = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeCollectionOfObjectValues<Account>("accounts", this.accounts);
        writer.writeCollectionOfObjectValues<Tag>("hashtags", this.hashtags);
        writer.writeCollectionOfObjectValues<Status>("statuses", this.statuses);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the statuses property value. The statuses property
     * @returns a Status
     */
    public get statuses() {
        return this._statuses;
    };
    /**
     * Sets the statuses property value. The statuses property
     * @param value Value to set for the statuses property.
     */
    public set statuses(value: Status[] | undefined) {
        this._statuses = value;
    };
}

File: clients/csharp/sdk/Models/SearchResult.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace MastodonClientLib.Models {
    public class SearchResult : IAdditionalDataHolder, IParsable {
        /// <summary>The accounts property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<Account>? Accounts { get; set; }
#nullable restore
#else
        public List<Account> Accounts { get; set; }
#endif
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>The hashtags property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<Tag>? Hashtags { get; set; }
#nullable restore
#else
        public List<Tag> Hashtags { get; set; }
#endif
        /// <summary>The statuses property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<Status>? Statuses { get; set; }
#nullable restore
#else
        public List<Status> Statuses { get; set; }
#endif
        /// <summary>
        /// Instantiates a new SearchResult and sets the default values.
        /// </summary>
        public SearchResult() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static SearchResult CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new SearchResult();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"accounts", n => { Accounts = n.GetCollectionOfObjectValues<Account>(Account.CreateFromDiscriminatorValue)?.ToList(); } },
                {"hashtags", n => { Hashtags = n.GetCollectionOfObjectValues<Tag>(Tag.CreateFromDiscriminatorValue)?.ToList(); } },
                {"statuses", n => { Statuses = n.GetCollectionOfObjectValues<Status>(Status.CreateFromDiscriminatorValue)?.ToList(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteCollectionOfObjectValues<Account>("accounts", Accounts);
            writer.WriteCollectionOfObjectValues<Tag>("hashtags", Hashtags);
            writer.WriteCollectionOfObjectValues<Status>("statuses", Statuses);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}
